src/main/groovy/cruise/umple/UmplePlugin.groovy

```diff
@@ -5,16 +5,29 @@ import cruise.umple.UmpleConsoleConfig
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 
-class UmplePlugin implements Plugin<Project> {
+class UmpleGradlePlugin implements Plugin<Project> {
 
     private UmpleConsoleConfig cfg
 
     @Override
     void apply(final Project project) {
 
-
-        project.task('hello') << {
-            println "Hello, World!"
+        project.task('compileUmpleFile') << {
+			UmpleConsoleMain(cfg)
+		
+			if(project.hasProperty("umpleArgs"))
+			{
+				// arguments are specified through gradle by -P, separated by commas (-P is for project properties)
+				// eg: "gradle compileUmpleFile 'P-umpleArgs=test.ump,-g,Java'"
+				args(umpleArgs.split(','))
+				
+				UmpleConsoleMain(args)
+				runConsole()
+			}
+			else
+			{
+				println "Error: Command line arguments are required to compile an Umple file"
```

Comment:
Check what Gradle uses for errors, it's probably some exception that you're supposed to throw.